ui/app/components/inference/InputSnippet.tsx

```diff
@@ -27,7 +27,14 @@ function renderContentBlock(block: ResolvedInputMessageContent, index: number) {
   switch (block.type) {
     case "text": {
       if (typeof block.value === "object") {
-        return <TextMessage key={index} label="Text (Arguments)" content={JSON.stringify(block.value, null, 2)} type="structured" />;
+        return (
+          <TextMessage
+            key={index}
+            label="Text (Arguments)"
+            content={JSON.stringify(block.value, null, 2)}
```

Comment:
I know this isn't changed but we should probably be wrapping JSON.stringify to validate the type and doing it in the function body so we can handle errors if an unserializable object is passed. Not a blocker but we should create a separate issue for this.

Reply:
the type I believe should be validated upstream but we can open an issue for this